### ISSUE TITLE
support dart 2 ?

### ISSUE THREAD

**Issue by lyquocnam** [angel-dart/angel#61]

**Comment by thosakwe** [comment]
Yes, Angel supports Dart 2, And has since Dart 2.0.0-dev versions.

Did you run into any trouble?

**Comment by lyquocnam** [comment]
get this error when install angel-cli ` pub global activate angel_cli`

```bash
Resolving dependencies...
The current Dart SDK version is 2.0.0.

Because angel_cli <1.3.0 requires SDK version <2.0.0 and angel_cli >=1.3.0 depends on inflection ^0.4.1, every version
  of angel_cli requires inflection ^0.4.1.
So, because inflection >=0.2.0 requires SDK version <2.0.0 and pub global activate depends on angel_cli any, version
  solving failed.
```

**Comment by nroberson** [comment]
I'm getting this error as well. 
tldr - a few of angel_cli's dependencies are not compatible with Dart 2.0.0 since they do not specify a sdk version constraint. This was not mandatory in 2.0.0-dev.

I found this note towards the end of this page https://www.dartlang.org/dart-2

> Packages must have an upper constraint of <3.0.0 to work in Dart 2 stable and subsequent Dart 2 releases. Dart 2 dev builds before the stable release have lax upper constraint checking and can use packages that have no SDK constraints or an upper constraint of <2.0.0.

I was able resolve the issue with inflection by adding the sdk constraint to inflection's pubspec and specifying the local path to inflection in the angel_cli dependencies. However, this led to the next dependency with the same issue, [homedir](https://github.com/kasperpeulen/homedir.dart). This led to recase (they're already fixed theirs), and then to mustache4dart. I think you get the idea.

So until those upstream dependencies are taken care of you'll need to revert back to 2.0.0-dev.

**Comment by thosakwe** [comment]
Correct; until Angel 2 is released (I’ve been working on it everyday),
Angel 1 users will need to stick to Dart 2 dev versions.

My intention is to send PR’s to the maintainers of the external
dependencies; otherwise, I will have to reimplement the functionality
myself.

**Comment by thosakwe** [comment]
Currently blocked on https://github.com/gmosx/dart-inflection/pull/13.

**Comment by thosakwe** [comment]
Closing, as Angel 2 has been in the works for some time now.
